distributed/stealing.py

```diff
@@ -223,13 +234,26 @@ def recalculate_cost(self, ts: TaskState) -> None:
 
     def put_key_in_stealable(self, ts: TaskState) -> None:
         cost_multiplier, level = self.steal_time_ratio(ts)
-        if cost_multiplier is not None:
-            assert level is not None
-            assert ts.processing_on
-            ws = ts.processing_on
-            worker = ws.address
-            self.stealable[worker][level].add(ts)
-            self.key_stealable[ts] = (worker, level)
+
+        prefix = ts.prefix
+        duration = self.scheduler._get_prefix_duration(prefix)
+        if cost_multiplier is None:
+            return
+
+        assert level is not None
+        assert ts.processing_on
+        ws = ts.processing_on
+        worker = ws.address
+        self.stealable[worker][level].add(ts)
+        self.key_stealable[ts] = (worker, level)
+
+        if duration == ts.prefix.duration_average:
+            return
+
+        s = self.unknown_durations.get(prefix.name)
+        if s is None:
```

Comment:
is this equivalent to ``if s not in self.unknown_durations``? If yes, that would be easier to read in the future

Reply:
I guess this could be rewritten, I've just moved that bit.

Reply:
Adjusted.